yt/fields/field_info_container.py

```diff
@@ -402,6 +404,12 @@ def add_output_field(self, name, sampling_type, **kwargs):
         kwargs.setdefault("ds", self.ds)
         self[name] = DerivedField(name, sampling_type, NullFunc, **kwargs)
 
+    def __setitem__(self, key, value):
+        ftype, fname = key
+        if any(fname == _fname for _ftype, _fname in self):
+            self._ambiguous_field_names[fname].add(ftype)
+        super().__setitem__(key, value)
+
```

Comment:
I'd suggest adding the method
```suggestion

     def is_ambiguous_field_name(self, fname):
         """Detect ambiguous field names
         
         Parameters
         ------------
         fname : str, the field name to check
         
         Returns
         --------
         True if the field is ambiguous, False otherwise. Ambiguous fields
         are fields with the same field name but different field types.
         """
         return fname in self._ambiguous_field_names
 
```